### PR TITLE
Avoid timeouts on test and generate in travis

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

`make generate` , `make goverals` and `make test` can sometimes don't print anything for minutes if running on a weak system. Protect these commands with printing something every minute.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,8 @@ install:
 - git reset --hard
 
 script:
-- make generate
+  # adding a keep-alive to avoid timeouts if there is no output for 10 minutes
+- while sleep 9m; do echo "keep alive" & make generate
```
Comment (davidvossel) [inline .travis.yml]: don't you need a `done` somewhere in here for the while loop?

**Comment by rmohr** [comment]
@murilofv  would you mind having a look at the ppc64 build issues on travis?

**Comment by murilofv** [comment]
> @murilofv would you mind having a look at the ppc64 build issues on travis?

Sure, I think part of the problem that Power is running slow is that I had to set 
```
build:ppc64le --jobs=1
run:ppc64le --jobs=1
```

For the `ci.bazelrc` in the `hack/dockerized` file due to the error seen in this comment where Bazel would end abruptly: 
https://github.com/kubevirt/kubevirt/pull/2944#issuecomment-576769015

So it may be why commands have been failing to complete within the time limit of Travis output expectation. I need to revisit this error to see if there is a better way to workaround the issue without slowing it too much as it currently seems to be the case, maybe playing around with other args like `--ram_utilization_factor` and `--local_ram_resources` instead of `--jobs`.

Alternatively, should we just run the keep alive for the whole test in Power?
Move the following to the beginning of the `.travis.yml` file and only run it on Power:
```
- while sleep 9m; do echo "keep alive"; done 
```

Might pollute the logs a bit, but it would ensure we are not failing due to a command running too long.

**Comment by rmohr** [comment]
> Alternatively, should we just run the keep alive for the whole test in Power?
> Move the following to the beginning of the `.travis.yml` file and only run it on Power:
> 
> ```
> - while sleep 9m; do echo "keep alive"; done 
> ```
> 
> Might pollute the logs a bit, but it would ensure we are not failing due to a command running too long.

That does not really work, because every `-` is a new subshell and background commands are just terminated.

My issue was also that even when I had `keep alive` properly working, that it printed something after the first 9 minutes, but nothing was printed afterwards anymore, a little bit like the whole build process was really stuck.

**Comment by murilofv** [comment]
> > Alternatively, should we just run the keep alive for the whole test in Power?
> > Move the following to the beginning of the `.travis.yml` file and only run it on Power:
> > ```
> > - while sleep 9m; do echo "keep alive"; done 
> > ```
> > 
> > 
> > Might pollute the logs a bit, but it would ensure we are not failing due to a command running too long.
> 
> That does not really work, because every `-` is a new subshell and background commands are just terminated.
> 
> My issue was also that even when I had `keep alive` properly working, that it printed something after the first 9 minutes, but nothing was printed afterwards anymore, a little bit like the whole build process was really stuck.

Do you have an example of when this happened so I could take a look? I did a quick test and it seems to print properly: https://travis-ci.com/github/murilofv/kubevirt/jobs/301819916

See lines 1097, 1125 and 1205 for the keep alive messages. The problem I see with this approach though is that if any command indeed hangs the build would keep running "indefinitely" (until possibly the timeout of 50 min that Travis has for a job). 

I still need to check about removing the --jobs=1 from Bazel to see if I can get it to run faster without running into the issue I mentioned previously.

**Comment by rmohr** [comment]
> Do you have an example of when this happened so I could take a look? I did a quick test and it seems to print properly: https://travis-ci.com/github/murilofv/kubevirt/jobs/301819916

Maybe it was my messed up travis patch. It would be great if it continues. I don't have the build anymore because I retriggered the build.

**Comment by murilofv** [comment]
I tried the `--ram_utilization_factor` and `--local_ram_resources`, but they didn't work, I still get bazel ending abruptly with them, so far the only thing that worked out is the keep alive.

**Pull request closed without merging**